src/app/utilities/transmart-utilities/transmart-constraint-mapper.ts

```diff
@@ -22,13 +22,23 @@ import {Study} from '../../models/constraint-models/study';
 import {Concept} from '../../models/constraint-models/concept';
 import {TrialVisit} from '../../models/constraint-models/trial-visit';
 import {TransmartConstraintSerialiser} from './transmart-constraint-serialiser';
-import {ConstraintHelper} from '../constraint-utilities/constraint-helper';
+import {ConstraintMark} from '../../models/constraint-models/constraint-mark';
 
 export class TransmartConstraintMapper {
 
   static fullSerialiser = new TransmartConstraintSerialiser(true);
   static defaultSerialiser = new TransmartConstraintSerialiser(false);
 
+  public static mapConstraintOnPatientLevel(constraint: Constraint, full?: boolean): object {
```

Comment:
I wonder in what cases this is needed.

Reply:
do you mean in general for which calls we need to wrap constraint in patient subselection or that this wrapping could be done differently ?

Reply:
Yes, in general. If, for instance, you would do a counts call for:
1. a biomaterial query, like all biomaterials of all twins;
2. the same query, but then wrapped in a patient selection.
The patient count will give the same result for both (the observation count may be different, though).